cachi2/core/utils.py

```diff
@@ -74,3 +77,30 @@ def load_json_stream(s: str) -> Iterator:
     while match := non_whitespace.search(s, i):
         obj, i = decoder.raw_decode(s, match.start())
         yield obj
+
+
+def copy_directory(origin: Path, destination: Path) -> Path:
+    """
+    Recursively copy directory to another path.
+
+    Use reflinks by default if the file system supports it.
+
+    :raise FileExistsError: if the destination path already exists.
+    :raise FileNotFoundError: if the origin directory does not exist.
+    """
+
+    def copy_using(copy_function: Callable) -> None:
```

Comment:
Can we prefix the name with a _ to make it more obvious later that `copy_using` isn't a library function?